raiden/ui/startup.py

```diff
@@ -0,0 +1,363 @@
+import sys
+from typing import Any, Dict, NamedTuple, Optional, Tuple
+
+import click
+from eth_utils import to_canonical_address, to_checksum_address
+from web3 import Web3
+
+from raiden.constants import Environment, RoutingMode
+from raiden.exceptions import AddressWithoutCode, AddressWrongContract, ContractVersionMismatch
+from raiden.network.blockchain_service import BlockChainService
+from raiden.network.discovery import ContractDiscovery
+from raiden.network.pathfinding import configure_pfs
+from raiden.network.proxies import (
+    SecretRegistry,
+    ServiceRegistry,
+    TokenNetworkRegistry,
+    UserDeposit,
+)
+from raiden.network.throttle import TokenBucket
+from raiden.network.transport import UDPTransport
+from raiden.settings import DEVELOPMENT_CONTRACT_VERSION, RED_EYES_CONTRACT_VERSION
+from raiden.utils.typing import Address
+from raiden_contracts.constants import (
+    CONTRACT_ENDPOINT_REGISTRY,
+    CONTRACT_SECRET_REGISTRY,
+    CONTRACT_SERVICE_REGISTRY,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    CONTRACT_USER_DEPOSIT,
+    ID_TO_NETWORKNAME,
+)
+from raiden_contracts.contract_manager import (
+    contracts_precompiled_path,
+    get_contracts_deployment_info,
+)
+
+from .sync import check_discovery_registration_gas
+
+
+def environment_type_to_contracts_version(environment_type: Environment) -> str:
+    if environment_type == Environment.DEVELOPMENT:
+        contracts_version = DEVELOPMENT_CONTRACT_VERSION
+    else:
+        contracts_version = RED_EYES_CONTRACT_VERSION
+
+    return contracts_version
+
+
+def setup_network_id_or_exit(
+        config: Dict[str, Any],
```

Comment:
Please do not do this, the function only uses `config` to set a key:

```python
config['chain_id'] = given_network_id
```

the configuration is a dictionary with lots of keys that has quite a few dependencies during app setup, it's completely unnecessary to pass it to a simple function because of one key.

Reply:
When you make a suggestion can you simply say what you want to suggest instead of how wrong the current code is?
So you don't like passing the config by reference so that a key's value is set. What is your alternative suggestion?